packages/scheduler/src/__tests__/SchedulerBrowser-test.js

```diff
@@ -178,6 +180,66 @@ describe('SchedulerBrowser', () => {
     runtime.assertLog(['Message Event', 'Continuation']);
   });
 
+  it('continuations should yield after deadline', () => {
+    function expensiveWork() {
+      runtime.log(`Continuation at ${performance.now()}ms`);
+
+      if (performance.now() >= 400) {
+        runtime.log(`Task finished at ${performance.now()}ms`);
+        return null;
+      }
+
+      runtime.advanceTime(100);
+      return expensiveWork;
+    }
+
+    scheduleCallback(UserBlockingPriority, () => {
+      runtime.log(`Task at ${performance.now()}ms`);
+      return expensiveWork;
+    });
+
+    runtime.assertLog(['Post Message']);
+
+    runtime.fireMessageEvent();
+    runtime.assertLog([
+      'Message Event',
+      'Task at 0ms',
+      'Continuation at 0ms',
+      'Post Message',
+    ]);
+
+    runtime.fireMessageEvent();
+    runtime.assertLog([
+      'Message Event',
+      'Continuation at 100ms',
+      'Post Message',
+    ]);
+
+    runtime.fireMessageEvent();
+    runtime.assertLog([
+      'Message Event',
+      'Continuation at 200ms',
+      'Post Message',
+    ]);
+
+    // The original task has expired and the browser wants us to yield back
+    // for other work. Schedule the continuation back to the browser to free up
+    // the event loop.
+    runtime.fireMessageEvent();
+    runtime.assertLog([
+      'Message Event',
+      'Continuation at 300ms',
```

Comment:
Without this change, continuations can cause an infinite loop after the task has expired.

I think if users of the scheduler need to "flush" work after it expires, they should flush it themselves instead of posting continuations that are immediately run.